App/Components/CreateThreadModal.tsx

```diff
@@ -53,8 +69,19 @@ class CreateThreadComponent extends React.Component<
       }
       this.setState({ submitted: true })
       this.props.completeScreen(this.state.value)
+      const threadConfig = {
+        type:
+          this.props.type !== undefined ? this.props.type : Thread.Type.OPEN,
+        sharing:
+          this.props.sharing !== undefined
+            ? this.props.sharing
+            : Thread.Sharing.SHARED,
+        whitelist:
+          this.props.whitelist !== undefined ? this.props.whitelist : [],
+        name: this.state.value
+      }
```

Comment:
this constructing of the ThreadConfig in the view Component might set of Aaron alarms, where he might suggest that it get it's own Saga to handle the transformation before attempting to create the thread. I'm okay with it for now though! :)